lib/Elevate/Blockers/Databases.pm

```diff
@@ -40,6 +46,71 @@ sub _warning_if_postgresql_installed ($self) {
     return 2;
 }
 
+sub _blocker_acknowledge_postgresql_datadir ($self) {
+
+    return 0 unless Cpanel::Pkgr::is_installed('postgresql-server');
+
+    my $touch_file = POSTGRESQL_ACK_TOUCH_FILE;
+    return 0 if -e $touch_file;
+
+    my @users_with_dbs = $self->_has_mapped_postgresql_dbs();
+    return 0 unless scalar @users_with_dbs;
+
+    my $message = <<~"EOS";
+    One or more users on your system have associated PostgreSQL databases.
+    ELevate may upgrade the software packages associated with PostgreSQL
+    automatically, but if it does, it will *NOT* automatically update the
+    PostgreSQL data directory to work with the new version. Without an update
+    to the data directory, the upgraded PostgreSQL software will not start, in
+    order to ensure that your data does not become corrupted.
+
+    For more information about PostgreSQL upgrades, please consider the
+    following resources:
+
+    https://cpanel.github.io/elevate/blockers/#postgresql
+    https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/using-databases#migrating-to-a-rhel-8-version-of-postgresql_using-postgresql
+    https://www.postgresql.org/docs/10/pgupgrade.html
+
+    When you are ready to acknowledge that you have prepared to update the
+    PostgreSQL data directory, or that this warning does not apply to you,
+    please touch the following file to continue with the ELevate process:
+
+    > touch $touch_file
+
+    The following user(s) have PostgreSQL databases associated with their cPanel accounts:
+    EOS
+
+    $message .= join "\n", sort(@users_with_dbs);
+
+    return $self->has_blocker($message);
+}
+
+sub _has_mapped_postgresql_dbs ($self) {
+
+    # Normalize locale:
+    local $ENV{HTTP_COOKIE} = 'session_locale=en';
+
+    # Get a list of cPanel users:
+    my $users_json   = Cpanel::SafeRun::Simple::saferunnoerror(qw(/usr/local/cpanel/bin/whmapi1 --output=json list_users));
+    my $users_parsed = Cpanel::JSON::Load($users_json);
+    die "WHM API1 list_users call returned result 0: $users_parsed->{metadata}->{reason}" unless $users_parsed->{metadata}->{result};
+    my @users = grep { $_ ne "root" } $users_parsed->{data}->{users}->@*;
+
+    # Compile a list of users with PgSQL DBs:
+    my @users_with_dbs;
+    foreach my $user (@users) {
+        my $dbs_json   = Cpanel::SafeRun::Simple::saferunnoerror( "/usr/local/cpanel/bin/uapi", "--user=$user", qw(--output=json Postgresql list_databases) );
```

Comment:
Why do we need to know if the databases are owned by cPanel? corrupt is corrupt. Can't we just check if any psql databases exist?

Reply:
Yes, we can check if anything other than the default `postgres`, `template0`, and `template1` databases exist.

> corrupt is corrupt.

There never was any corruption, AFAICT.

Reply:
I just meant that making the "controversial" uapi calls could be avoided :D